File: backend/app/models/lesson.rb
class Lesson < ApplicationRecord
  belongs_to :course

  validates :title, presence: true, length: { minimum: 3 }
  validates :status, inclusion: { in: %w[draft published] }

  after_initialize :set_default_status, if: :new_record?

  private

  def set_default_status
    self.status ||= "draft"
  end
end

File: backend/src/models/Lesson.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Lesson = sequelize.define('Lesson', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false,
    validate: { len: [3, 255] },
  },
  status: {
    type: DataTypes.STRING,
    allowNull: false,
    defaultValue: 'draft',
    validate: { isIn: [['draft', 'published']] },
  },
  video_url: {
    type: DataTypes.STRING,
  },
  course_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
}, {
  tableName: 'lessons',
  underscored: true,
});

module.exports = Lesson;
